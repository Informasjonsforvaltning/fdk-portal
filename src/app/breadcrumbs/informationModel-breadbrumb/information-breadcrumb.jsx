import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { resolve } from 'react-resolver';
import { getTranslateText } from '../../../lib/translateText';
import { getInformationmodel } from '../../../api/informationmodels';

const memoizedGetInformationModel = _.memoize(getInformationmodel);

export const PureInformationModelBreadcrumb = props => {
  const { informationModelItem } = props;
  return <span>{getTranslateText(_.get(informationModelItem, 'title'))}</span>;
};

const mapProps = {
  informationModelItem: props =>
    memoizedGetInformationModel(props.match.params.id)
};

PureInformationModelBreadcrumb.defaultProps = {
  informationModelItem: null
};

PureInformationModelBreadcrumb.propTypes = {
  informationModelItem: PropTypes.object
};

export const InformationModelBreadcrumb = resolve(mapProps)(
  PureInformationModelBreadcrumb
);
