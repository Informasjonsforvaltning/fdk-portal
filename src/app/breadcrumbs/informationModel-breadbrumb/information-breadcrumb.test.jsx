import React from 'react';
import { shallow } from 'enzyme';
import { PureInformationModelBreadcrumb } from './information-breadcrumb';

test('should render PureInformationModelBreadcrumb with "Dataset" active correctly', () => {
  const defaultProps = {
    informationModelItem: {
      title: 'api title'
    }
  };
  const wrapper = shallow(<PureInformationModelBreadcrumb {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render PureInformationModelBreadcrumb correctly with no informationModelItem', () => {
  const defaultProps = {};
  const wrapper = shallow(<PureInformationModelBreadcrumb {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
