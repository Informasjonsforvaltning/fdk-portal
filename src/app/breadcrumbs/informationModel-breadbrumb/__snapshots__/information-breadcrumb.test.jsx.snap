// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PureInformationModelBreadcrumb correctly with no informationModelItem 1`] = `<span />`;

exports[`should render PureInformationModelBreadcrumb with "Dataset" active correctly 1`] = `
<span>
  api title
</span>
`;
